browser/request_otr/request_otr_browsertest.cc

```diff
@@ -0,0 +1,425 @@
+  infobar_manager->AddObserver(&observer);
+
+  SetRequestOTRPref(RequestOTRService::RequestOTRActionOption::kAsk);
+  GURL url = embedded_test_server()->GetURL("sensitive.a.com", "/simple.html");
+  NavigateTo(url);
+  ASSERT_TRUE(IsShowingInterstitial());
+
+  // Simulate click on 'Proceed Normally' button. This should navigate to
+  // the originally requested page in off-the-record mode.
+  ClickAndWaitForNavigation("back-button");
+  ASSERT_FALSE(IsShowingInterstitial());
+  // Request-OTR infobar should never appear, because the user requested to
+  // proceed normally, so we should not be in off-the-record mode.
+
+  infobar_manager->RemoveObserver(&observer);
+}
+
+IN_PROC_BROWSER_TEST_F(RequestOTRBrowserTest, IncludeExclude) {
+  ASSERT_TRUE(InstallMockExtension());
+  SetRequestOTRPref(RequestOTRService::RequestOTRActionOption::kAsk);
+  GURL url1 = embedded_test_server()->GetURL("www.b.com", "/simple.html");
+  NavigateTo(url1);
+  ASSERT_TRUE(IsShowingInterstitial());
+  GURL url2 =
+      embedded_test_server()->GetURL("notsensitive.b.com", "/simple.html");
+  NavigateTo(url2);
+  ASSERT_FALSE(IsShowingInterstitial());
+}
+
+IN_PROC_BROWSER_TEST_F(RequestOTRBrowserTest, HistoryAfterStandardNavigation) {
+  ASSERT_EQ(GetHistoryCount(), 0);
+  NavigateTo(
+      embedded_test_server()->GetURL("notsensitive.b.com", "/simple.html"));
+  ASSERT_EQ(GetHistoryCount(), 1);
+}
+
+IN_PROC_BROWSER_TEST_F(RequestOTRBrowserTest, HistoryAfterOTRNavigation) {
+  ASSERT_TRUE(InstallMockExtension());
+
+  ASSERT_EQ(GetHistoryCount(), 0);
+  SetRequestOTRPref(RequestOTRService::RequestOTRActionOption::kAlways);
+  NavigateTo(embedded_test_server()->GetURL("sensitive.a.com", "/simple.html"));
+  ASSERT_EQ(GetHistoryCount(), 0);
+}
+
+IN_PROC_BROWSER_TEST_F(RequestOTRBrowserTest,
+                       WindowOpenAfterStandardNavigation) {
+  NavigateTo(embedded_test_server()->GetURL("sensitive.a.com", "/simple.html"));
+  ASSERT_TRUE(content::ExecJs(
+      web_contents(), "window.open('notsensitive.b.com/simple.html');"));
+  ASSERT_NE(content::EvalJs(web_contents(), "window.opener"), nullptr);
+}
+
+IN_PROC_BROWSER_TEST_F(RequestOTRBrowserTest, WindowOpenAfterOTRNavigation) {
+  ASSERT_TRUE(InstallMockExtension());
+
+  // Always use request-otr for sensitive sites (skipping interstitial).
+  SetRequestOTRPref(RequestOTRService::RequestOTRActionOption::kAlways);
+
+  NavigateTo(embedded_test_server()->GetURL("sensitive.a.com", "/simple.html"));
+  ASSERT_TRUE(content::ExecJs(
+      web_contents(), "window.open('notsensitive.b.com/simple.html');"));
+  ASSERT_EQ(content::EvalJs(web_contents(), "window.opener"), nullptr);
+}
+
+class RequestOTRDisabledBrowserTest : public RequestOTRBrowserTestBase {
+ public:
+  RequestOTRDisabledBrowserTest() {
+    feature_list_.InitAndDisableFeature(kBraveRequestOTR);
+  }
+
+ private:
+  base::test::ScopedFeatureList feature_list_;
+};
+
+class RequestOTR1PESBrowserTest : public RequestOTRBrowserTest {
```

Comment:
How is `RequestOTR1PESBrowserTest` different from `RequestOTRBrowserTest`? Shouldn't service workers always be disabled for RequestOTR? It's not clear to me what the 1PES part is about here and why it's separate from the other `RequestOTRBrowserTest` tests

Reply:
added comments to explain